cirq-web/cirq_web/bloch_sphere/bloch_sphere.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+import webbrowser
+import uuid
+
+from typing import Union
+from pathlib import PosixPath, WindowsPath
+
+from numpy import ndarray
+
+from cirq_web import widget
+
+from cirq.testing import random_superposition
+from cirq.qis import to_valid_state_vector
+from cirq.qis.states import bloch_vector_from_state_vector
+from cirq.protocols import to_json
+
+
+class BlochSphere(widget.Widget):
+    def __init__(
+        self,
+        sphere_radius: int = 5,
+        state_vector: ndarray = to_valid_state_vector([math.sqrt(2) / 2, math.sqrt(2) / 2]),
+        random: bool = False,
+    ):
+        """Initializes a BlochSphere, gathering all the user information and
+        converting to JSON for output.
+
+        Also initializes it's parent class Widget with the bundle file provided.
+
+        Args:
+            sphere_radius: the radius of the bloch sphere in the three.js diagram.
+            The default value is 5.
+
+            state_vector: a state vector to pass in to be represented. The default
+            vector is 1/sqrt(2) (|0⟩ +|1⟩) (the plus state)
+        """
+
+        super().__init__('cirq_ts/dist/bloch_sphere.bundle.js')
+
+        self.sphere_json = self._convert_sphere_input(sphere_radius)
```

Comment:
This is a bigger change, but will simplify the code a bunch - I don't think we need the JSON layer here. We can just directly set the arguments in the function signature for a given object. My bad, I recommended JSON at the start of the project, but in fact JSON will be only required when we want to represent more complicated objects to pass down (maybe in the 3D Circuit case).

Reply:
Done